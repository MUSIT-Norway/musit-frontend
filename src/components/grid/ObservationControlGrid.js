

import React, { Component, PropTypes } from 'react'
import { Table, FormGroup } from 'react-bootstrap'
import FontAwesome from 'react-fontawesome'
import { hashHistory } from 'react-router'

export default class ObservationControlGrid extends Component {
  static propTypes = {
    id: PropTypes.number.isRequired,
    translate: PropTypes.func.isRequired,
    tableData: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      type: PropTypes.string.isRequired,
      doneDate: PropTypes.string.isRequired,
      types: PropTypes.object.isRequired,
      doneBy: PropTypes.string.isRequired,
      registeredDate: PropTypes.string.isRequired,
      registeredBy: PropTypes.string.isRequired
    }))
  }
  static iconMap = {
    temperature: 'asterisk',
    inertAir: 'cloud',
    relativeHumidity: 'tint',
    cleaning: 'fa',
    lightCondition: 'sun-o',
    alchohol: 'percent',
    gas: 'inr',
    mold: 'bolt',
    pest: 'bug',
    envdata: 'truck'
  }

  render() {
    const { id, translate } = this.props
    const showEnabledIcon = (data, type) => {
      return (
        (data) ? <FontAwesome style={{ padding: '2px' }} name={ObservationControlGrid.iconMap[type]} /> : ''
      )
    }
    const showDisabledIcon = (data, type) => {
      return (
        (data === false) ?
          <FontAwesome style={{ color: 'gray', padding: '2px' }} name={ObservationControlGrid.iconMap[type]} /> : ''
      )
    }
    return (
      <FormGroup>
        <div>
          <Table responsive hover condensed>
            <thead>
              <tr>
                <th />
                <th>
                  {translate('musit.grid.observation.date')}
                </th>
                <th>
                  {translate('musit.grid.observation.types')}
                </th>
                <th>
                  {translate('musit.grid.observation.doneBy')}
                </th>
                <th>
                  {translate('musit.grid.observation.registeredDate')}
                </th>
                <th>
                  {translate('musit.grid.observation.registeredBy')}
                </th>
              </tr>
            </thead>
            <tbody>
              {this.props.tableData.map((c, i) =>
                <tr
                  key={i}
                  id={`${c.id}_${c.doneDate}`}
                  onClick={() =>
                    (hashHistory.push(c.type === 'control' ?
                    `magasin/${id}/control/${c.id}` : `magasin/${id}/observation/${c.id}`
                  ))}
                >
                  <td id={`${c.id}_${c.doneDate}_type`}>
                    {c.type && c.type.toLowerCase() === 'control' ? <FontAwesome name="user-secret" /> : ''}
                    {c.type && c.type.toLowerCase() === 'observation' ? <FontAwesome name="eye" /> : ''}
                  </td>
                  <td id={`${c.id}_${c.doneDate}_date`}>
                    {`${c.doneDate}`}
                  </td>
                  <td id={`${c.id}_${c.doneDate}_types`}>
                    {showEnabledIcon(c.types.ControlTemperature, 'temperature')}
                    {showDisabledIcon(c.types.ControlTemperature, 'temperature')}
                    {showEnabledIcon(c.types.ControlHypoxicAir, 'inertAir')}
                    {showDisabledIcon(c.types.ControlHypoxicAir, 'inertAir')}
                    {showEnabledIcon(c.types.ControlRelativeHumidity, 'relativeHumidity')}
                    {showDisabledIcon(c.types.ControlRelativeHumidity, 'relativeHumidity')}
                    {showEnabledIcon(c.types.ControlCleaning, 'cleaning')}
                    {showDisabledIcon(c.types.ControlCleaning, 'cleaning')}
                    {showEnabledIcon(c.types.ControlLightingCondition, 'lightCondition')}
                    {showDisabledIcon(c.types.ControlLightingCondition, 'lightCondition')}
                    {showEnabledIcon(c.types.ControlAlcohol, 'alchohol')}
                    {showDisabledIcon(c.types.ControlAlcohol, 'alchohol')}
                    {showEnabledIcon(c.types.ControlGas, 'gas')}
                    {showDisabledIcon(c.types.ControlGas, 'gas')}
                    {showEnabledIcon(c.types.ControlMold, 'mold')}
                    {showDisabledIcon(c.types.ControlMold, 'mold')}
                    {showEnabledIcon(c.types.ControlPest, 'pest')}
                    {showDisabledIcon(c.types.ControlPest, 'pest')}
                    {showEnabledIcon(c.types.envdata, 'envdata')}
                    {showDisabledIcon(c.types.envdata, 'envdata')}
                  </td>
                  <td id={`${c.id}_${c.doneDate}_doneBy`}>
                    {`${c.doneBy}`}
                  </td>
                  <td id={`${c.id}_${c.doneDate}_registeredDate`}>
                    {`${c.registeredDate}`}
                  </td>
                  <td id={`${c.id}_${c.doneDate}_registeredBy`}>
                    {`${c.registeredBy}`}
                  </td>
                </tr>
              )}
            </tbody>
          </Table>
        </div>
      </FormGroup>
    )
  }
}
