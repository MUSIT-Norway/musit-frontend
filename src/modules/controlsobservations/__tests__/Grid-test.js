import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import React from 'react';
import ObservationControlGrid from '../EventsGrid';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('ObservationControlGrid', () => {
  it('Check the 1st row Date value', () => {
    const myDiv = mount(
      <ObservationControlGrid
        id="1"
        translate={key => key}
        showControl={() => true}
        showObservation={() => true}
        tableData={[
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            lightingCondition: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: undefined,
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            temperature: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            hypoxicAir: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            relativeHumidity: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            cleaning: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            mold: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            alcohol: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            gas: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            waterDamageAssessment: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            fireProtection: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            theftProtection: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            perimeterSecurity: {
              ok: true
            }
          },
          {
            id: 1,
            doneDate: '2017-01-01T09:42:52+00:00',
            doneBy: 'Blablabla...',
            registeredDate: '2017-01-04T09:42:57+00:00',
            registeredBy: 'Blabla...',
            eventType: 'control',
            pest: {
              ok: true
            }
          }
        ]}
      />
    );
    expect(mountToJson(myDiv)).toMatchSnapshot();
  });
});
