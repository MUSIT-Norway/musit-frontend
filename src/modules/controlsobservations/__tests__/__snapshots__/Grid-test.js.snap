// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationControlGrid Check the 1st row Date value 1`] = `
<ObservationControlGrid
  id="1"
  showControl={[Function]}
  showObservation={[Function]}
  tableData={
    Array [
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "lightingCondition": Object {
          "ok": true,
        },
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": undefined,
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
        "temperature": Object {
          "ok": true,
        },
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "hypoxicAir": Object {
          "ok": true,
        },
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
        "relativeHumidity": Object {
          "ok": true,
        },
      },
      Object {
        "cleaning": Object {
          "ok": true,
        },
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "mold": Object {
          "ok": true,
        },
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "alcohol": Object {
          "ok": true,
        },
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "gas": Object {
          "ok": true,
        },
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
        "waterDamageAssessment": Object {
          "ok": true,
        },
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "fireProtection": Object {
          "ok": true,
        },
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
        "theftProtection": Object {
          "ok": true,
        },
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "perimeterSecurity": Object {
          "ok": true,
        },
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
      Object {
        "doneBy": "Blablabla...",
        "doneDate": "2017-01-01T09:42:52+00:00",
        "eventType": "control",
        "id": 1,
        "pest": Object {
          "ok": true,
        },
        "registeredBy": "Blabla...",
        "registeredDate": "2017-01-04T09:42:57+00:00",
      },
    ]
  }
  translate={[Function]}
>
  <FormGroup
    bsClass="form-group"
  >
    <div
      className="form-group"
    >
      <div>
        <Table
          bordered={false}
          bsClass="table"
          condensed={true}
          hover={true}
          responsive={true}
          striped={false}
        >
          <div
            className="table-responsive"
          >
            <table
              className="table table-condensed table-hover"
            >
              <thead>
                <tr>
                  <th />
                  <th>
                    Dato
                  </th>
                  <th>
                    Typer
                  </th>
                  <th>
                    Utf√∏rt av
                  </th>
                  <th>
                    Reg. dato
                  </th>
                  <th>
                    Reg. av
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="0"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitlightingcondicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Lysforhold"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="1"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musittemperatureicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Temperatur"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  />
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="2"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musithypoxicairicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Inertluft"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="3"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitrelhumidityicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Relativ luftfuktighet"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="4"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitcleaningicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Renhold"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="5"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitmoldicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Mugg"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="6"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitalcoholicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Sprit"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="7"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitgasicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Gass"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="8"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitwaterdamageicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Vannskaderisiko"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="9"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitfireprotectionicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Brannsikring"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="10"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musittheftprotectionicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Tyverisikring"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="11"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitperimetersecurityicon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Skallsikring"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
                <tr
                  id="1_2017-01-01T09:42:52+00:00"
                  key="12"
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                >
                  <td
                    id="1_2017-01-01T09:42:52+00:00_type"
                  >
                    <div
                      className="icon icon-musitcontrolicon"
                      title="Kontroll"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_date"
                  >
                    01.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_types"
                  >
                    <span
                      className="icon icon-musitpesticon"
                      key="6"
                      style={
                        Object {
                          "color": "darkgray",
                          "padding": "2px",
                        }
                      }
                      title="Skadedyr"
                    />
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_doneBy"
                  >
                    Blablabla...
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredDate"
                  >
                    04.01.2017
                  </td>
                  <td
                    id="1_2017-01-01T09:42:52+00:00_registeredBy"
                  >
                    Blabla...
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </Table>
      </div>
    </div>
  </FormGroup>
</ObservationControlGrid>
`;
